src/util/amount.rs

```diff
@@ -863,6 +863,35 @@ impl ::std::iter::Sum for SignedAmount {
     }
 }
 
+/// Calculate the sum over the iterator using checked arithmetic.
+pub trait CheckedSum<R> {
```

Comment:
Can we restrict this further as follows? That way, this also matches the usage and description. 

> pub trait CheckedSum<R> : Iterator <Item = R>

Reply:
Sure, I just wasn't really sure what the benefit would be (it's just a trait because it has to so I can extend other structs). On a second thought I'm more inclined to seal the trait.

Reply:
+1 for making it Sealed

Reply:
Sealed it.